modules/javafx.graphics/src/main/native-glass/win/PlatformSupport.h

```diff
@@ -61,14 +61,19 @@ class PlatformSupport final
      * Collect the specified platform preferences and notify the JavaFX application when a preference has changed.
      * The change notification includes all specified preferences, not only the changed preferences.
      */
-    bool updatePreferences(PreferenceType) const;
+    bool updatePreferences(PreferenceType, bool delayedChangesExpected) const;
 
     /**
      * Handles the WM_SETTINGCHANGE message.
-    */
+     */
     bool onSettingChanged(WPARAM, LPARAM) const;
 
 private:
+    /**
+     * Suggested aggregation delay for changes that come in over a period of time.
+     */
+    static constexpr int SUGGESTED_DELAY_MILLIS = 1000;
```

Comment:
1 second seems too long.
What is the typical range for the high contrast train of changes?

Reply:
Quite a lot, Windows takes around 1-2 seconds to change to and from a high-contrast theme (there's even a full-frame wait screen that pops up). The suggested delay of 1 second works reliably on my machine.

Reply:
If it takes that long to update all the colors, maybe we should not add complexity to this subsystem and just send the events as they come?

What exactly is the problem the user experiences right now?

Reply:
Are you sure a delay is a good way to solve this?  It won't guarantee that no inconsistent states will be observed (what if Windows is slower, or a change is reverted within the delay period -- you may still see "half" changes).

Reply:
I don't think that there's a perfect way to solve this. Yes, this can theoretically happen, and then you'll see a visual glitch. I can imagine that this might also be the reason why Windows shows a full-screen popup when you switch to and from a high-contrast theme: to hide glitchy applications during the transition period.

Reply:
Alright, just making sure, as you mentioned in another thread seeing inconsistent transient states isn't an acceptable trade-off.  They can still occur, and so users can't rely on them.  In that respect the situation is similar to before this change, other than it being less likely to occur (and perhaps never or rarely in practice).

Reply:
Yes, I think having inconsistent transient states is not an acceptable trade-off, that's why I think reducing them by 99,9% (from guaranteed to happen, to almost hever happening) is sensible. If it does happen anyway, it shouldn't crash your application but show up as a visual glitch (I mean, we're talking about colors). In that sense, you can "rely" on it.